lib/wasmer-types/src/units.rs

```diff
@@ -108,9 +109,16 @@ where
     }
 }
 
-impl From<Bytes> for Pages {
-    fn from(bytes: Bytes) -> Self {
-        Self((bytes.0 / WASM_PAGE_SIZE) as u32)
+const PAGES_EXCEED_UINT32: &'static str = "Number of pages exceeds uint32 range";
+
+impl TryFrom<Bytes> for Pages {
+    type Error = &'static str;
+
+    fn try_from(bytes: Bytes) -> Result<Self, Self::Error> {
+        let pages: u32 = (bytes.0 / WASM_PAGE_SIZE)
+            .try_into()
+            .or(Err(PAGES_EXCEED_UINT32))?;
```

Comment:
I believe you can write:

```suggestion
            .try_into()
            .map_err(PAGES_EXCEED_UINT32)?;
```

Reply:
`map_err` takes a callback, which is what I wanted to avoid. It would have to be

```rust
.map_err(|_original_error| PAGES_EXCEED_UINT32)?;
```

However, we can use this as well. Which way do you like more?